index.d.ts

```diff
@@ -247,15 +247,33 @@ export type Observer<T> = {
   next?(value: T): void
 }
 
+/**
+ * Extend the state
+ *
+ * This is used by store enhancers and store creators to extend state.
+ * If there is no state extension, it just returns the state, as is, otherwise
+ * it returns the state joined with its extension.
+ */
+export type ExtendState<State, Extension> = [Extension] extends [never]
+  ? State
+  : State & Extension
+
 /**
  * A store is an object that holds the application's state tree.
  * There should only be a single store in a Redux app, as the composition
  * happens on the reducer level.
  *
  * @template S The type of state held by this store.
  * @template A the type of actions which may be dispatched by this store.
+ * @template StateExt any extension to state from store enhancers
+ * @template Ext any extensions to the store from store enhancers
  */
-export interface Store<S = any, A extends Action = AnyAction> {
+export interface Store<
+  S = any,
+  A extends Action = AnyAction,
+  StateExt = never,
+  Ext = {}
```

Comment:
Hey, would it be possible to make `StateExt` and `Ext` optional whenever not using an Enhancer?
At the moment `createStore<S,A,StateExt,Ext>` generic typings are required as opposed as `createStore<S,A>` for stores without enhancers. Thanks!